### PR TITLE
feat:  add custom error messages for chainId errors

### PR DESCRIPTION
**Pull request by iykazrji**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [x] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on enhancing error handling and user feedback for wallet connection processes in the application. It introduces custom error messages, modifies connection logic, and ensures proper disconnection procedures.

### Detailed summary
- Added `customErrorMessages` for wallet connection errors in `strings.ts`.
- Introduced `CustomErrorMessage` type in `types.ts`.
- Updated `useConnectEOA` hook for better error handling.
- Implemented `clearWalletConnectStore` function in `disconnect.ts`.
- Enhanced error handling in `ConnectionError` component.
- Refactored `EoaConnectCard` and `WalletConnectCard` to utilize new error handling logic.
- Removed redundant connection logic from `EoaPickCard` and streamlined wallet connection handling.
- Added utility function `getErrorMessage` for generating error messages based on specific conditions.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: account-kit/react/src/components/auth/card/eoa.tsx [hunk 1]
```diff
@@ -51,25 +61,49 @@ export const EoaConnectCard = () => {
...
 };
 
 export const WalletConnectCard = () => {
   const { setAuthStep, authStep } = useAuthContext("wallet_connect");
+  const { walletConnectParams } = useWalletConnectAuthConfig();
+  const { chain } = useChain();
+  const signer = useSigner();
+  const walletConnectConnector = walletConnectParams
+    ? walletConnect(walletConnectParams)
+    : null;
```
Comment (moldy530) [inline account-kit/react/src/components/auth/card/eoa.tsx:81]: we actually might want to throw if this connector is ever `null` this card should never be rendered without the params for wallet connect being present

File: account-kit/react/src/components/auth/card/eoa.tsx [hunk 2]
```diff
@@ -204,3 +201,20 @@ export const EoaPickCard = () => {
...
+  },
+};
+
+const getCustomErrorMessage = (error: Error) => {
+  if (error.message.includes("ChainId not found")) {
+    return customErrorMessages.chainIdNotFound;
+  }
+
+  // Use default error message
+  return null;
```
Comment (moldy530) [inline account-kit/react/src/components/auth/card/eoa.tsx:219]: I think we should have the default error message live here and maybe rename this method to `getErrorMessageCopy` or something similar

File: account-kit/react/src/components/auth/card/error/connection-error.tsx [hunk 3]
```diff
@@ -20,8 +21,29 @@ export const ConnectionError = ({
...
+    if (
+      connectionType === "wallet" &&
+      EOAConnector === EOAWallets.WALLET_CONNECT
+    ) {
+      clearWalletConnectStore();
+    }
+
+    if (signer) {
+      signer.disconnect();
+    }
```
Comment (moldy530) [inline account-kit/react/src/components/auth/card/error/connection-error.tsx:39]: we should use the `import { disconnect } from "@account-kit/core";` or does this cause some weird stuff to happen?

File: account-kit/react/src/components/auth/card/error/connection-error.tsx [hunk 4]
```diff
@@ -74,15 +100,36 @@ export const ConnectionError = ({
...
+    const store = txn.objectStore("keyvaluestorage");
+
+    // Clear Store
+    store.clear();
+  };
+
+  dbOpenRequest.onerror = () => {
+    console.error("Error opening Wallet Connect DB. Cannot clear store.");
+  };
+};
```
Comment (moldy530) [inline account-kit/react/src/components/auth/card/error/connection-error.tsx:135]: this feels like something that should live in the `import { disconnect } from "@account-kit/core";` function instead

File: account-kit/react/src/components/auth/card/eoa.tsx [hunk 5]
```diff
@@ -51,25 +61,49 @@ export const EoaConnectCard = () => {
       error={authStep.error}
       secondaryButton={{
         title: "Cancel",
-        onClick: () => setAuthStep({ type: "initial" }),
+        onClick: async () => {
+          // Ensure to stop all inflight requests
+          await signer?.disconnect();
```
[the remark this answers is not in the document]
Comment (iykazrji) [reply account-kit/react/src/components/auth/card/eoa.tsx]: This has been changed to remove the disconnection logic here as it really isn't necessary at this stage.

**Review by rthomare — APPROVED** [review]

**Pull request merged**